tests/subsys/net/openthread/rpc/client/src/cmsng_suite.c

```diff
@@ -197,4 +197,115 @@ ZTEST(ot_rpc_commissioning_cli, test_discover_cb_handler_empty)
 	zassert_true(cb_called);
 }
 
+/* Test serialization of otDatasetSetActive() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActive)
+{
+	otError error;
+
+	const otOperationalDataset dataset = {
+		{0x0123456789abcdefull, 0x1234, false},
+		{0x0123456789abcdefull, 0x1234, false},
+		{{INT_SEQUENCE(OT_NETWORK_KEY_SIZE)}},
+		{{INT_SEQUENCE(OT_NETWORK_NAME_MAX_SIZE)}},
+		{{INT_SEQUENCE(OT_EXT_PAN_ID_SIZE)}},
+		{{INT_SEQUENCE(OT_IP6_PREFIX_SIZE)}},
+		0x12345678ul,
+		0xabcd,
+		0xef67,
+		{{INT_SEQUENCE(OT_PSKC_MAX_SIZE)}},
+		{0x9876, 1, 0, 1, 0, 1, 0, 1, 0, 1, 7},
+		0xfedcba98,
+		{false, true, false, true, false, true, false, true, false, true, false, true}};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_SET_ACTIVE, DATASET),
+				   RPC_RSP(OT_ERROR_NONE));
+	error = otDatasetSetActive(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+}
+
+/* Test incoming parameters validation of otDatasetSetActive() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActive_negative)
+{
+	otError error;
+
+	error = otDatasetSetActive(NULL, NULL);
+	zassert_equal(error, OT_ERROR_INVALID_ARGS);
+}
+
+/* Test serialization of otDatasetGetActive() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetGetActive)
+{
+	otError error;
+	otOperationalDataset dataset;
+	uint8_t net_key[] = {INT_SEQUENCE(OT_NETWORK_KEY_SIZE)};
+	uint8_t nwk_name[] = {INT_SEQUENCE(OT_NETWORK_NAME_MAX_SIZE), 0};
+	uint8_t ext_pan_id[] = {INT_SEQUENCE(OT_EXT_PAN_ID_SIZE)};
+	uint8_t local_prefix[] = {INT_SEQUENCE(OT_IP6_PREFIX_SIZE)};
+	uint8_t pskc[] = {INT_SEQUENCE(OT_PSKC_MAX_SIZE)};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_GET_ACTIVE), RPC_RSP(DATASET));
+	error = otDatasetGetActive(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+	zassert_equal(dataset.mActiveTimestamp.mSeconds, 0x0123456789abcdefull);
+	zassert_equal(dataset.mActiveTimestamp.mTicks, 0x1234);
+	zassert_false(dataset.mActiveTimestamp.mAuthoritative);
+
+	zassert_equal(dataset.mPendingTimestamp.mSeconds, 0x0123456789abcdefull);
+	zassert_equal(dataset.mPendingTimestamp.mTicks, 0x1234);
+	zassert_false(dataset.mPendingTimestamp.mAuthoritative);
+
+	zassert_mem_equal(dataset.mNetworkKey.m8, net_key, OT_NETWORK_KEY_SIZE);
+	zassert_mem_equal(dataset.mNetworkName.m8, nwk_name, OT_NETWORK_NAME_MAX_SIZE + 1);
+	zassert_mem_equal(dataset.mExtendedPanId.m8, ext_pan_id, OT_EXT_PAN_ID_SIZE);
+	zassert_mem_equal(dataset.mMeshLocalPrefix.m8, local_prefix, OT_IP6_PREFIX_SIZE);
+	zassert_equal(dataset.mDelay, 0x12345678ul);
+	zassert_equal(dataset.mPanId, 0xabcd);
+	zassert_equal(dataset.mChannel, 0xef67);
+	zassert_mem_equal(dataset.mPskc.m8, pskc, OT_PSKC_MAX_SIZE);
+
+	zassert_equal(dataset.mSecurityPolicy.mRotationTime, 0x9876);
+	zassert_true(!!dataset.mSecurityPolicy.mObtainNetworkKeyEnabled);
```

Comment:
nit: These are already bools, so don't need `!!`.

Reply:
Hm... This is bitfield: https://github.com/nrfconnect/sdk-nrfxlib/blob/main/openthread/include/openthread/dataset.h#L153

Reply:
yes, but `bool` bitfield so you can compare it directly with `false` or `true`.

Reply:
I'm not sure if this is correct, since C standard doesn't guarantee true is always 1.
![Screenshot from 2024-08-06 08-49-02](https://github.com/user-attachments/assets/a9092451-3779-4288-920f-87c621c5beac)

Reply:
This is obsolete behavior. Nevertheless, it is not possible to exclude on 100%

Reply:
`zassert` macros don't compare the condition with `true`. They compare to `false`. But this doesn't matter much. Disregard this suggestion.